ws-server/src/tests.rs

```diff
@@ -569,3 +587,58 @@ async fn run_forever() {
 	// Send the shutdown request from one handle and await the server on the second one.
 	join(server_handle.clone().stop().unwrap(), server_handle).with_timeout(TIMEOUT).await.unwrap();
 }
+
+#[tokio::test]
+async fn unsubscribe_twice_should_indicate_error() {
+	init_logger();
+	let addr = server().await;
+	let mut client = WebSocketTestClient::new(addr).with_default_timeout().await.unwrap().unwrap();
+
+	let sub_call = call("subscribe_hello", Vec::<()>::new(), Id::Num(0));
+	let sub_id: u64 = deser_call(client.send_request_text(sub_call).await.unwrap());
+
+	let unsub_call = call("unsubscribe_hello", vec![sub_id], Id::Num(1));
+	let unsub_1: String = deser_call(client.send_request_text(unsub_call).await.unwrap());
+	assert_eq!(&unsub_1, "Unsubscribed");
+
+	let unsub_call = call("unsubscribe_hello", vec![sub_id], Id::Num(2));
+	let unsub_2 = client.send_request_text(unsub_call).await.unwrap();
+	let unsub_2_err: RpcError = serde_json::from_str(&unsub_2).unwrap();
+	let sub_id = to_json_raw_value(&sub_id).unwrap();
+	assert_eq!(
+		unsub_2_err,
+		RpcError {
+			jsonrpc: v2::TwoPointZero,
+			error: v2::ErrorObject {
+				code: v2::ErrorCode::ServerError(v2::error::INVALID_SUBSCRIPTION_CODE),
+				message: v2::error::INVALID_SUBSCRIPTION_MSG,
+				data: Some(&to_json_raw_value(&format!("ID={} is not active", sub_id)).unwrap()),
+			},
```

Comment:
We should have a macro for converting a code to an ErrorObject, `to_error_object!(INVALID_SUBSCRIPTION_CODE)`

Reply:
we already have a `From<ErrorCode> for ErrorObject` lemme see I can make this a little less ugly